src/postamble.js

```diff
@@ -225,7 +225,7 @@ function run(args) {
   if (preloadStartTime === null) preloadStartTime = Date.now();
 
   if (runDependencies > 0) {
-#if ASSERTIONS
+#if ASSERTIONS && !BINARYEN_ASYNC_COMPILATION
```

Comment:
Maybe I misunderstand: when running the GCC torture tests the waterfall's JS files contain this printErr:
https://storage.googleapis.com/wasm-llvm/builds/linux/18611/wasm-torture-emwasm-18611.tbz2

IIUC ASSERTIONS isn't set?
https://github.com/WebAssembly/waterfall/blob/master/src/build.py#L1101